test-integration/replication.test.js

```diff
@@ -32,36 +30,22 @@ const data = [
     uri: 'mongodb://localhost:27002'
   }
 ]
-const servers = []
-
-for (let index = 0; index < data.length; index += 1) {
-  const { hostname, uri } = data[index]
-  const client = new MongoClient(uri)
-  const server = new MongoServerState(hostname, client)
-  data[index].server = server
-  servers.push(server)
-}
+const servers = data.map((settings) => new ReplicaMemberMongoServerState(settings))
 
 test.serial('ReplicaSetState can pick up uninitialized state', async (t) => {
-  const configuration = new ReplicationConfiguration({ replicaSetName, servers })
-
-  const state = new ReplicaSetState(configuration)
+  const state = new ReplicaSetState(servers, { replicaSetName })
```

Comment:
Why did this change?  This is similar  to what I did here https://github.com/unumotors/mongo-hydra/compare/cli-config-file?expand=1#diff-0a97ecae3217f05b7a3c91d9b6627a02  

What happens if we just support that as an input? i.e it takes an object as configuration.